### ISSUE TITLE
Automatically read newest report files from specified location

### ISSUE THREAD

**Issue by anxiousmodernman** [ahelium/monthlyrep#5]
Right now you're hardcoding paths here:

https://github.com/ahelium/monthlyrep/blob/master/report_configs/Jan_2015.py#L1

It would be better to just run ```mrep.py combined``` with no other arguments and have it read the latest report files from a pre-determined location.

```
/some-folder
  /open-reports
  /click-reports
  /profile-reports
```

It's probably cool to keep the behavior of ```mrep.py unsubs``` (and others) the same, taking a filename as the next argument.

Once this is done, the only config you should need will be the location of ```some-folder```.

**Comment by anxiousmodernman** [comment]
Make a new branch for this. Once your master is up to date, run

```
$ git checkout -b read-newest-5
```

**Comment by anxiousmodernman** [comment]
@ahelium where are you going to store the path of ```some-folder``` (top level one)

**Comment by ahelium** [comment]
https://github.com/ahelium/monthlyrep/blob/master/mrep.py

Remind me what the config portion of config['unsub'] etc... is doing and where it is coming from please.

**Comment by anxiousmodernman** [comment]
It is coming from here:
https://github.com/ahelium/monthlyrep/blob/master/mrep.py#L8

```REPORT_FILES``` is your config object. Imported straight from some location.

You pass it, along with comman line args here:
https://github.com/ahelium/monthlyrep/blob/master/mrep.py#L82

**Comment by anxiousmodernman** [comment]
```REPORT_FILES``` is currently where you're getting that path. But the filename that it refers to doesn't need to be month-specific anymore. So you'll need a new file or a renamed file to store the config of ```some-folder```. That's all I mean.

**Comment by anxiousmodernman** [comment]
You wrote:
 
>Ok. I think I need to go in a different direction here. I think I need to use os, but i need to use it to get all the df's in a folder, and process them all each month. I think this needs to happen so that the output can easily be pasted into excel without manually updating too much

Basically. But try not to always think in terms of pandas dataframes. That's too low level and we should be able to work with the classes contained in the ```report``` package.

Here's my idea. Given these folders...

```
/some-folder
  /open-reports
  /click-reports
  /profile-reports
```

...write some "service" or "utility" class that can give you the **paths to the most recent report files**. So when you make a new instance of this class, you can ask it for the paths. So you can have code that looks like this.

```python
from settings import REPORTS_FOLDER  # hardcode path to top-level folder (string)
from report.service import ReportPathUtility
from report.monthly import Profile, Opens, Unsubscribes

path_util = ReportPathUtility(REPORTS_FOLDER) # constructed w/ path to top-level folder, maybe?

path_to_latest_opens = path_util.get_latest_open_report_path() # returns full path as string
opens_report = Opens(path_to_latest_opens)  # we can construct our Opens with a file path, right?

path_to_latest_unsubs = path_util.get_latest_unsub_report_path() # ditto. another method on ReportPathUtility
unsubs_report = Unsubs(path_to_latest_unsubs)

# etc...

```

**Comment by anxiousmodernman** [comment]
You wrote:

> Question is, this seems kind of repetitive. Like if i used this 'make_file_dict' function, i will need to pass both get directory and get latest to it. Is this what you had in mind at all? Or am I off the mark.

I assume you are referring to the ```make_report_files_dict``` method.

Two things. And be advised that I am **making an assumption** about what you want to do, because I think I told you to do it at some point. :)

First, you would **not** need to pass anything to ```make_report_files_dict```. This is the beauty of object oriented programming. You can make your dict like this (read the comments!):

```python
def make_report_files_dict(self):  # removed path
    unsub_dir = self.get_unsub_directory()  # access instance methods with self
    profile_dir = self.get_profile_directory()
    opens_dir = self.get_open_rate_directory()

    report_files = dict()  # initialize empty dict, alternative to report_files = {}
    report_files['unsubs'] = self.get_latest_file(unsub_dir)       # Set your dict values here.
    report_files['profile'] = self.get_latest_file(profile_dir)    # NOTE: self (the instance of the class) is
    report_files['opens'] = self.get_latest_file(opens_dir)        # passed implicitly for non-static methods

    return report_files
```

I didn't actually run that, so a copy and paste might fail at some point. But that is the general idea.

Second, you are right about redundancy from the perspective of the **user** or **consumer** of this class. 

The best outcome is for the consumer of this class to not care at all about the details of file path manipulation, or where the reports are. The consumer simply says, "Hey, Service. Gimme the filepaths for the latest reports." And the service should do that. That's the idea of **encapsulation** in OOP. 

Read my comments, keep going, and make your tests pass. When they do, we can make this even more awesome.

**Comment by ahelium** [comment]
Ugh now im so confused. It looks like the repo has gone back in time? Like,
before, there were 3 sub folders in the 'tests/test_toplevel_folder', now i
only see one? I stashed all the chagnes I made on my work computer and
pulled it down with

git pull upstream read-newest-5

**Comment by anxiousmodernman** [comment]
You are viewing the master branch. Click the drop down menu and switch the
branches on GitHub

**Comment by ahelium** [comment]
No! I already switched the branches. And if i was that wouldnt explain the fact that I dont see the things I created in pycharm when i pull read-newest-5. Right?

**Comment by anxiousmodernman** [comment]
You clicked the drop down on GitHub?

**Comment by ahelium** [comment]
Yep, i can see some of the updates i made to services, but somehow not the test files.

**Comment by anxiousmodernman** [comment]
Maybe you didn't push them.

I saw a one commit against the test file I think

**Comment by ahelium** [comment]
```python
# code code code
```

(3:40:40 PM) Coleman McFarland: (backticks, on the ~ key)
(3:41:18 PM) Coleman McFarland: (use the preview, too, for help)

**Comment by anxiousmodernman** [comment]
@ahelium is this issue done?
